src/system/Form/Radio/Radio.tsx

```diff
@@ -0,0 +1,97 @@
+/** @jsxImportSource theme-ui */
+
+import classNames from 'classnames';
+import React, { Ref, forwardRef, useCallback } from 'react';
+import { ThemeUIStyleObject } from 'theme-ui';
+
+import { RadioOption } from './RadioOption';
+
+export const VIP_RADIO = 'vip-radio-component';
+
+export type RadioOptionOptionProps = {
+	id: string;
+	value: string;
+	disabled?: boolean;
+	className?: string;
+	label?: string;
+	renderLabel?: ( props, labelStyles: ThemeUIStyleObject ) => JSX.Element | null;
+	labelProps?: object;
+	inputProps?: object;
+};
+
+export interface RadioOptionProps {
+	option: RadioOptionOptionProps;
+	name: string;
+	variant: string;
+	disabled: boolean | undefined;
+	onChangeHandler: ( e: React.ChangeEvent< HTMLInputElement > ) => void;
+	checked: boolean;
+}
+
+export type RadioProps = {
+	variant?: string;
+	disabled?: boolean | undefined;
+	defaultValue?: string | number;
+	onChange?: ( e: React.ChangeEvent< HTMLInputElement >, option?: RadioOptionOptionProps ) => void;
+	name?: string;
+	options?: RadioOptionOptionProps[];
+	className?: string;
+};
+
+const Radio = forwardRef< HTMLDivElement, RadioProps >(
```

Comment:
personally i think we should reevaluate using `forwardRef` everywhere. who needs a ref on a `<div>`?